utility/include/isobus/utility/thread_synchronization.hpp

```diff
@@ -38,6 +106,95 @@ namespace isobus
 /// @param x The mutex to lock.
```

Comment:
Nit: extra unneeded `;` on line 130. It's not part of the diff but it makes a lot of angry warnings with pedantic warnings enabled